### ISSUE TITLE
"RequireJS.setup(...)" doesn't work in play 2.4.0-RC1

### ISSUE THREAD

**Issue by djx314** [webjars/webjars-play#52]
In play 2.4.0-RC1, reverse routers takes constructor params, so RequireJS.setup("javascripts/lib/archerui/") in template throws an exception like:
```scala
play.api.http.HttpErrorHandlerExceptions$$anon$1: Execution exception[[InstantiationException: controllers.ReverseWebJarAssets]]
...
Caused by: java.lang.NoSuchMethodException: controllers.ReverseWebJarAssets.<init>()
```

I suggest that just get the reverse route like this:
```scala
//RequireJS.scala
def nastyReflectedRoute(path: String, routerName: String): Call = {
  val clazz = Class.forName("controllers.routes", true, play.api.Play.current.classloader)
  val field = clazz.getDeclaredField(routerName)
  val routeInstance = field.get(null)
  routeInstance.asInstanceOf[{ def at(path: String): Call }].at(path)
}
```
and it works well in my project.

**Comment by jamesward** [comment]
I wonder if for 2.4 we can actually use DI instead of reflection?  Seems like the Reverse Router should be able to be injected.  /cc @jroper

**Comment by jroper** [comment]
Reverse routers *can* be injected, but no bindings are provided for them by default.  Regardless, you would still have to use reflection - the reverse router classes simply *do not exist* at the time that this library is compiled, and therefore, there's nothing that they can be compiled against.

In Play 3, we're getting rid of global state, and that means we're going to have to rethink how assets are served, we're probably going to have to have some assets components, with a route that gets injected, and an injectable reverse router specifically for assets that is not generated but rather provided by Play.  That will solve this issue.

Anyway, the suggested fix is what should be done to address this.

**Comment by jamesward** [comment]
#53 has the PR.  Only thing I'm not sure about is why the paths are being escaped:
https://github.com/webjars/webjars-play/blob/fix-reflection/src/test/scala/org/webjars/requirejs/RequireJSSpec.scala#L15

Is that a problem?

**Comment by jroper** [comment]
In your routes file, you have:

    GET     /webjars/:file       controllers.WebJarAssets.at(file)

`:file` only matches a single path segment, so when you put a value that contains slashes in there, they get escaped.  If you want to match multiple path segments, you have to use `*file`.

**Comment by jamesward** [comment]
Oh man.  Right.  I'm on vacation so I have a good excuse for this.  PR updated.

**Comment by jamesward** [comment]
I've released `2.4.0-RC2` with this fix.
